### PR TITLE
Add search text box and filtering capability to Launcher profile assets editor

### PR DESCRIPTION
**Pull request by GPayne**
It can be hard to find an asset in the profile editor since there are many assets, and the tree view has collapsed branches by default for directories that don't contain any currently-selected assets.
This adds a text box above the tree view to allow for easy searching of the available asset files.

### CODE REVIEW

**Comment by WeirdRubberDuck** [comment]
What Qt version are you using? I can't compile this code using Qt 6.4.0

When trying to compile this, I got problems with the compiler not being able to find `QRegExp`. Did some quick research and this seems to have been removed in Qt6, and should seemingly be replaced with the `QRegularExpression` class. 

https://doc.qt.io/qt-6/qtcore-changes-qt6.html
https://doc.qt.io/qt-6/qregexp.html#porting-to-qregularexpression

**Comment by alexanderbock** [comment]
QRegExp is deprecated in Qt6; there is a page on how to convert the old usage:  https://doc.qt.io/qt-6/qregexp.html#porting-to-qregularexpression

**Comment by GPayne** [comment]
Latest commit now uses `QRegularExpression`. The code works with both Qt6 and Qt5 now.

File: apps/OpenSpace/ext/launcher/include/profile/assetsdialog.h [hunk 1]
```diff
@@ -27,13 +27,14 @@
 
 #include <QDialog>
 #include <QSortFilterProxyModel>
-#include <QRegExp>
+#include <QRegularExpression>
```
Comment (alexanderbock) [inline apps/OpenSpace/ext/launcher/include/profile/assetsdialog.h:30]: Should be sorted before the `QSortFilterProxyModel`

File: apps/OpenSpace/ext/launcher/include/profile/assetsdialog.h [hunk 2]
```diff
@@ -26,12 +26,39 @@
...
 
+class SearchProxyModel : public QSortFilterProxyModel {
+Q_OBJECT
+public:
+    /**
+     * Constructor for SearchProxyModel class
+     *
+     * \param parent The QObject* object that is the Qt parent of this object
+     */
+    SearchProxyModel(QObject* parent = nullptr) : QSortFilterProxyModel(parent) {}
```
Comment (alexanderbock) [inline apps/OpenSpace/ext/launcher/include/profile/assetsdialog.h:44]: The definition of the constructor should move to the cpp file

File: apps/OpenSpace/ext/launcher/include/profile/assetsdialog.h [hunk 3]
```diff
@@ -26,12 +26,39 @@
...
+     * \param pattern The QString reference containing the regex pattern
+     */
+    void setFilterRegularExpression(const QString& pattern);
+
+protected:
+    bool filterAcceptsRow(int sourceRow, const QModelIndex& sourceParent) const override;
+
+private:
+    QRegularExpression* _regExPattern = nullptr;
+    bool acceptIndex(const QModelIndex& idx) const;
```
Comment (alexanderbock) [inline apps/OpenSpace/ext/launcher/include/profile/assetsdialog.h:59]:
```suggestion
    bool acceptIndex(const QModelIndex& idx) const;

    QRegularExpression* _regExPattern = nullptr;
```

In general, we sort functions first, then member variables

File: apps/OpenSpace/ext/launcher/src/profile/assetsdialog.cpp [hunk 4]
```diff
@@ -62,7 +62,6 @@ namespace {
     void traverseToFindFilesystemMatch(AssetTreeModel& model, QModelIndex parent,
                                        int nRows, const std::string& path)
     {
-
         int startIndex = 0;
         std::string token = "${USER_ASSETS}/";
         if (path.find(token) == 0) {
```
Comment (alexanderbock) [inline apps/OpenSpace/ext/launcher/src/profile/assetsdialog.cpp:67]: Can be replaced with `path.starts_with(token)`

File: apps/OpenSpace/ext/launcher/src/profile/assetsdialog.cpp [hunk 5]
```diff
@@ -259,3 +280,57 @@ void AssetsDialog::parseSelections() {
...
+        if (matchIterator.hasNext()) {
+            return true;
+        }
+        for (int row = 0; row < idx.model()->rowCount(idx); ++row) {
+            if (acceptIndex(idx.model()->index(row, 0, idx))) {
+                return true;
+            }
+        }
+    }
+    return false;
```
Comment (alexanderbock) [inline apps/OpenSpace/ext/launcher/src/profile/assetsdialog.cpp:335]:
```suggestion
    if (!idx.isValid() || !_regExPattern) {
        return false;
    }
    QString text = idx.data(Qt::DisplayRole).toString();
    QRegularExpressionMatchIterator matchIterator = _regExPattern->globalMatch(text);
    if (matchIterator.hasNext()) {
        return true;
    }
    for (int row = 0; row < idx.model()->rowCount(idx); ++row) {
        if (acceptIndex(idx.model()->index(row, 0, idx))) {
            return true;
        }
    }

    return false;
```

Would reduce overall indentation

**Review by alexanderbock — APPROVED** [review]

File: apps/OpenSpace/ext/launcher/src/profile/assetsdialog.cpp (hunk 5 above)
Comment (GPayne) [reply apps/OpenSpace/ext/launcher/src/profile/assetsdialog.cpp:335]: All changes made

**Review by alexanderbock — APPROVED** [review]

**Pull request merged**